src/init/CliRunner.ts

```diff
@@ -0,0 +1,57 @@
+import * as Path from 'path';
+import { ReadStream, WriteStream } from 'tty';
+import { Loader, LoaderProperties } from 'componentsjs';
+import yargs from 'yargs';
+import { RuntimeConfig } from './RuntimeConfig';
+import { Setup } from './Setup';
+
+/**
+ * Generic run function for starting the server from a given config
+ * @param args - Command line arguments.
+ * @param stdin - Standard input stream.
+ * @param stdout - Standard output stream.
+ * @param stderr - Standard error stream.
+ * @param properties - Components loader properties.
+ */
+export const runCustom = function(
+  args: string[],
+  stdin: ReadStream,
+  stdout: WriteStream,
+  stderr: WriteStream,
+  properties: LoaderProperties,
+): void {
+  const { argv } = yargs
+    .usage('node ./bin/server.js [args]')
+    .options({
+      port: { type: 'number', alias: 'p' },
+      config: { type: 'string', alias: 'c' },
+    })
+    .help();
+
+  new Promise<RuntimeConfig>(async(resolve): Promise<void> => {
+    // Load provided or default config file
+    const configPath = argv.config ?
+      Path.join(process.cwd(), argv.config) :
+      `${__dirname}/../../config/config-default.json`;
+
+    // Setup from config file
+    const loader = new Loader(properties);
+    await loader.registerAvailableModuleResources();
+    const setup: Setup = await loader
+      .instantiateFromUrl('urn:solid-server:my', configPath);
```

Comment:
@rubensworks Still a `my` here, is that a problem?

Reply:
Yes, it looks I didn't test things properly.
We probably want some automated tests around this bin...

Reply:
It's a class now, so we don't even need bin tests. But a system test that boots up the bin would be needed indeed.